include/ginkgo/core/reorder/reordering_base.hpp

```diff
@@ -81,36 +88,6 @@ struct ReorderingBaseArgs {
 };
 
 
-/**
- * Declares an Abstract Factory specialized for ReorderingBases
- */
-using ReorderingBaseFactory =
```

Comment:
public interface?

Reply:
I added a template argument to these two. But I think technically, the second one is still an interface break since in case someone used the `PolymorphicBase` template argument, this would no longer work...

Reply:
I'm fine if you break the interface at that point as I don't think a lot of people (my assumption is 0 outside of us) were using it. If it is compiled with `c++17`, the interface is no longer broken, so I would consider it as acceptable.